javascript/modules/client/src/client-dao.ts

```diff
@@ -90,44 +109,56 @@ export class ClientDaoWhitelist extends ClientCore
   // }
 }
 
-export class ClientDaoSimpleVote extends ClientCore
-  implements IClientDaoBase, IClientDaoSimpleVote {
+export class ClientDaoWhitelistVoting extends ClientCore
+  implements IClientDaoBase, IClientDaoWhitelistVoting {
   /** DAO related methods */
   dao = {
-    create: async (
-      _daoConfig: DaoConfig,
-      _tokenConfig: TokenConfig,
-      _mintConfig: MintConfig,
-      _votingConfig: VotingConfig,
-      _gsnForwarder?: string,
-    ): Promise<string> => {
-      // TODO: This is an ethers.js integration example
-      // const instance = this.attachContractExample(
-      //   "0x1234567890123456789012345678901234567890"
-      // );
-      // const tx = await instance.store("0x1234");
-      // await tx.wait();
+    create: async (params: ICreateDaoWhitelistVoting): Promise<string> => {
+      if (!this.signer)
+        throw new Error("A signer is needed for creating a DAO");
+      const daoFactoryInstance = DAOFactory__factory.connect(
+        this.daoFactoryAddress,
+        this.signer.connect(this.web3)
+      );
 
-      // TODO: Not implemented
-      return Promise.resolve("0x1234567890123456789012345678901234567890");
+      const registry = await daoFactoryInstance
+        .registry()
+        .then(registryAddress => {
+          return Registry__factory.connect(registryAddress, this.web3);
+        });
+
+      const newDaoRegisteredAddress = new DeferredPromise<string>();
```

Comment:
The event should come from the `tx.wait()` and not from the global registry contract events